pallets/compliance-manager/src/lib.rs

```diff
@@ -562,68 +536,96 @@ decl_event!(
 );
 
 impl<T: Trait> Module<T> {
+    /// Ensure that `origin` can modify `ticker`'s compliance rules.
+    fn ensure_can_modify_rules(
+        origin: T::Origin,
+        ticker: Ticker,
+    ) -> Result<IdentityId, DispatchError> {
+        let sender = ensure_signed(origin)?;
+        CallPermissions::<T>::ensure_call_permissions(&sender)?;
+        let did = Context::current_identity_or::<Identity<T>>(&sender)?;
```

Comment:
```suggestion
        let did = Identity::<T>::ensure_origin_call_permissions(&sender)?.primary_did;
```
We may not need to import `ensure_origin` or `Context`, or define `CallPermissions`.